action.yaml

```diff
@@ -48,32 +48,40 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
+
+    - name: Set up environment
+      run: uv venv
+      shell: bash
+
     - id: install-local-requirements
       if: ${{ inputs.requirements-file-paths != '' }}
       run: |
         IFS=',' read -ra requirements_paths <<< "${{ inputs.requirements-file-paths }}"
         for req in ${requirements_paths[@]}; do
-          pip install -r $req
+          uv pip install -r $req
         done
       shell: bash
 
     - id: install-from-pyproject-toml
       if: ${{ inputs.pyproject-toml-path != '' }}
       run: |
         cd $(dirname ${{ inputs.pyproject-toml-path }})
-        pip install .
+        uv pip install .
       shell: bash
 
     - id: prefect-deploy
       run: |
-        pip install prefect -U "prefect>=3,<4"
+        uv pip install prefect -U "prefect>=3,<4"
         if [ ${{ inputs.all-deployments }} == "true" ];
         then
-          prefect --no-prompt deploy --all --prefect-file "${{ inputs.deployment-file-path }}"
+          uv run prefect --no-prompt deploy --all --prefect-file "${{ inputs.deployment-file-path }}"
         else
           IFS=',' read -ra deployment_names <<< "${{ inputs.deployment-names }}"
           for name in "${deployment_names[@]}"; do
-            prefect --no-prompt deploy --prefect-file "${{ inputs.deployment-file-path }}" --name "$name"
+            uv run prefect --no-prompt deploy --prefect-file "${{ inputs.deployment-file-path }}" --name "$name"
```

Comment:
are these `uv run`s necessary? it [looks like](https://github.com/astral-sh/setup-uv) the setup-uv thing adds the new interpreter to the path already

Reply:
It adds uv to the path but does not activate the virtual environment. This makes `uv run` required I believe.